src/views/Rules/ListRules/index.jsx

```diff
@@ -441,13 +443,41 @@ function ListRules(props) {
   };
 
   const handleScroll = ({ scrollTop }) => {
-    if (listRef.current) {
-      listRef.current.scrollTo(scrollTop);
+    listRef.current.scrollTo(scrollTop);
+  };
+
+  const handleListScroll = ({ scrollOffset, scrollUpdateWasRequested }) => {
+    // scrollUpdateWasRequested is a boolean.
+    // This value is true if the scroll was caused by scrollToItem(),
+    // and false if it was the result of a user interaction in the browser.
+    if (scrollUpdateWasRequested) {
+      window.scrollTo(0, scrollOffset);
     }
   };
 
-  const handleListRef = component => {
-    listRef.current = component;
+  useEffect(() => {
+    if (hash !== ruleIdHash && filteredRulesCount) {
+      const ruleId = Number(hash.replace('#', '')) || null;
+
+      if (ruleId) {
+        const itemNumber = filteredRulesWithScheduledChanges
+          .map(rule => rule.rule_id)
+          .indexOf(ruleId);
+
+        listRef.current.scrollToItem(itemNumber, 'start');
+        setRuleIdHash(hash);
+      }
+    }
+  }, [hash, filteredRulesCount]);
```

Comment:
If I understand effects correctly, depending on `filteredRulesCount` will mean that this code runs anytime it changes (ie: whenever the drop down selection changes). If that's correct, what's the purpose of that?

Reply:
At first, `hash` will be defined but `filteredRulesCount` won't be there because the result depends on an asynchronous operation defined in another `useEffect`. The code will go inside this handler but won't go through the [if statement](https://github.com/mozilla-frontend-infra/balrog-ui/pull/57/files/d405b5f66ab1e59453cf6f9c0a50e1af966a0e5f#diff-6bdfeacc9dead52092c38bbdc3958037R459).

Next, when `filteredRulesCount` has been computed, the `useEffect` handler will run again but this time it will go pass the if statement since both `hash` and `filteredRulesCount` are defined.

When a drop down selection changes, the `useEffect` will trigger but it won't satisfy the if statement. The hash is removed when the drop down selection changes.

Hope that helps.

Reply:
Aaaah, I see. Thanks for the explanation.